src/Registry.sol

```diff
@@ -15,90 +17,171 @@ contract Registry is IRegistry {
+    function register(Registration[] calldata regs, address owner)
         external
         payable
         returns (bytes32 registrationRoot)
     {
+        // At least MIN_COLLATERAL for sufficient reward for fraud/equivocation challenges
         if (msg.value < MIN_COLLATERAL) {
             revert InsufficientCollateral();
         }
 
-        if (unregistrationDelay < MIN_UNREGISTRATION_DELAY) {
-            revert UnregistrationDelayTooShort();
-        }
-
+        // Construct tree root from registrations
         registrationRoot = _merkleizeRegistrations(regs);
 
         if (registrationRoot == bytes32(0)) {
             revert InvalidRegistrationRoot();
         }
 
+        // Prevent duplicates from overwriting previous registrations
         if (registrations[registrationRoot].registeredAt != 0) {
             revert OperatorAlreadyRegistered();
         }
 
+        // Each Operator is mapped to a unique registration root
         registrations[registrationRoot] = Operator({
-            withdrawalAddress: withdrawalAddress,
+            owner: owner,
             collateralGwei: uint56(msg.value / 1 gwei),
             registeredAt: uint32(block.number),
-            unregistrationDelay: unregistrationDelay,
             unregisteredAt: type(uint32).max,
             slashedAt: 0
         });
 
-        emit OperatorRegistered(registrationRoot, msg.value, unregistrationDelay);
+        emit OperatorRegistered(registrationRoot, uint56(msg.value / 1 gwei), owner);
     }
 
-    /// @notice Verify a merkle proof against a given `registrationRoot`
-    /// @dev The function will return the operator's collateral amount if the proof is valid or 0 if the proof is invalid.
+    /// @notice Starts the process to unregister an operator from the URC
+    /// @dev The function will mark the `unregisteredAt` timestamp in the Operator struct. The operator can claim their collateral after the `unregistrationDelay` more blocks have passed.
+    /// @dev The function will revert if:
+    /// @dev - The operator has already unregistered (AlreadyUnregistered)
+    /// @dev - The operator has not registered (NotRegisteredKey)
+    /// @dev - The caller is not the operator's withdrawal address (WrongOperator)
     /// @param registrationRoot The merkle root generated and stored from the register() function
-    /// @param leaf The leaf to verify
-    /// @param proof The merkle proof to verify the operator's key is in the registry
-    /// @param leafIndex The index of the leaf in the merkle tree
-    /// @return collateralGwei The collateral amount in GWEI
-    function verifyMerkleProof(bytes32 registrationRoot, bytes32 leaf, bytes32[] calldata proof, uint256 leafIndex)
-        external
-        view
-        returns (uint256 collateralGwei)
-    {
-        collateralGwei = _verifyMerkleProof(registrationRoot, leaf, proof, leafIndex);
+    function unregister(bytes32 registrationRoot) external {
```

Comment:
If an operator opts into a slasher, then unregisters without opting out of the slasher,  it seems like we will have a "dangling" slasher commitment inside `slasherCommitments`. Would that be a problem? I guess we _can_ say that the reader of `slasherCommitments` must also check `registrations` but I think that would be more error-prone.

Also related question: what was the issue with @AnshuJalan 's approach of storing the opt-in info inside `Operator`? Was there some cost issue?